guide/oauth2/README.md

```diff
@@ -170,35 +179,39 @@ What you did in the quick example was go through the `implicit grant` flow, whic
 
 #### Authorization code grant
 
-Unlike the quick example, you need an OAuth2 url where the `response_type` is `code`. Once you've obtained it, try visiting the link and authorizing your application. You should notice that instead of a hash, the redirect url now has a single query parameter appended to it like `?code=ACCESS_CODE`. Modify your `index.js` file to pull the parameter out of the url if it exists. You can use the `url` module to do this for us.
+Unlike the quick example, you need an OAuth2 url where the `response_type` is `code`. Once you've obtained it, try visiting the link and authorizing your application. You should notice that instead of a hash, the redirect url now has a single query parameter appended to it like `?code=ACCESS_CODE`. Modify your `index.js` (on `/` route) file to pull the parameter out of the url if it exists.
 
+Make the route as a async function:
 ```js
-const url = require('url');
-
-// ...
+app.get('/', async (request, response) => {
+	// ...
+});
+```
 
-const urlObj = url.parse(req.url, true);
+Then:
+<!-- eslint-skip -->
+```js
+let access_token;
+let token_type;
 
-if (urlObj.query.code) {
-	const accessCode = urlObj.query.code;
+if (request.query.code) {
+	const accessCode = request.query.code;
 	console.log(`The access code is: ${accessCode}`);
 }
 
-if (urlObj.pathname === '/') {
-	responseCode = 200;
-	content = fs.readFileSync('./index.html');
-}
+if (token_type && access_token && request.query.state) {
+	response.status(200).redirect(`/?code=${request.query.code}&state=${request.query.state}&access_token=${access_token}&token_type=${token_type}`);
+} else { response.sendFile('index.html', { root: '.' }); }
```

Comment:
Either make this an early return or place the `else {}` content on its own line.